DOCSUP-5932: Update the description of the position function normal (#21430)
ClickHouse/ClickHouse: docs/en/sql-reference/functions/string-search-functions.md

## position(haystack, needle), locate(haystack, needle) {#position}

> Searchs the substring `needle` in the string `haystack`.

Reviewer: ```suggestion
Searches for the substring `needle` in the string `haystack`.
```

Returns the position (in bytes) of the found substring in the string, starting from 1. 

For a case-insensitive search, use the function [positionCaseInsensitive](#positioncaseinsensitive).

**Syntax**

``` sql

> position((haystack, needle[, start_pos]) | (needle IN haystack))

Reviewer 2: It looks a bit weird and hard to read for users. Maybe it would be better to write two variants of syntax separately?

Author: > It looks a bit weird and hard to read for users. Maybe it would be better to write two variants of syntax separately?

Interesting idea. Also for dividing examples below.

``` 

Alias: `locate(haystack, needle[, start_pos])`.

!!! note "Note"

>     Syntax of position(needle IN haystack) provides SQL-compatibility and works the same way as position(haystack, needle).

Reviewer: ```suggestion
    Syntax of position(needle IN haystack) provides SQL-compatibility, the function works the same way as to position(haystack, needle).
```

**Arguments**

> -   `haystack` — string, in which substring will to be searched. [String](../../sql-reference/syntax.md#syntax-string-literal).

Reviewer: ```suggestion
-   `haystack` — String, in which substring will to be searched. [String](../../sql-reference/syntax.md#syntax-string-literal).
```

> -   `needle` — substring to be searched. [String](../../sql-reference/syntax.md#syntax-string-literal).

Reviewer: ```suggestion
-   `needle` — Substring to be searched. [String](../../sql-reference/syntax.md#syntax-string-literal).
```

> -   `start_pos` – Optional parameter, position of the first character in the string to start search. [UInt](../../sql-reference/data-types/int-uint.md).

Reviewer: ```suggestion
-   `start_pos` – Position of the first character in the string to start search. [UInt](../../sql-reference/data-types/int-uint.md). Optional.
```

└───────────────────────────────┘
```

> **Examples**

Reviewer 2: We already have this title above, why do we need it here again?

Author: Maybe we should put here something like this "Examples for POSITION(needle IN haystack) syntax"?
Or separation examples one from another syntax is not necessary?

Author: > We already have this title above, why do we need it here again?

Maybe we should put here something like this "Examples for POSITION(needle IN haystack) syntax"?
Or separation examples one from another syntax is not necessary?

Reviewer 2: > Maybe we should put here something like this "Examples for POSITION(needle IN haystack) syntax"?

Good idea.

> **For syntax (needle IN haystack)**

Reviewer: ```suggestion
**Examples**
```

Query:

```sql

> select 3 = position('c' in 'abc');

Reviewer: ```suggestion
SELECT 3 = position('c' IN 'abc');
```

└───────────────────────────────┘
```

**Examples**

> To see how the function position(needle IN haystack) works, check the queries below.

Reviewer: ```suggestion
```
